modules/externalipam/netbox/netbox_client.rb

```diff
@@ -0,0 +1,167 @@
+      params[:vrf_id] = get_group_id(group_name) if group_name
+      response = @api_resource.get("ipam/prefixes/", params)
+      json_body = JSON.parse(response.body)
+      return nil if json_body['count'].zero?
+      subnets_from_result(json_body['results'])
+    end
+
+    def ip_exists?(ip, subnet_id, group_name)
+      group_id = get_group_id(group_name)
+      params = { address: ip }
+      params[:prefix_id] = subnet_id unless subnet_id.nil?
+      params[:vrf_id] = group_id unless group_id.nil?
+      url = "ipam/ip-addresses/"
+      response = @api_resource.get(url, params)
+      json_body = JSON.parse(response.body)
+      !json_body['count'].zero?
+    end
+
+    def add_ip_to_subnet(ip, params)
+      desc = 'Address auto added by Foreman'
+      address = "#{ip}/#{params[:cidr].split('/').last}"
+      group_name = params[:group_name]
+      data = { address: address, nat_outside: 0, description: desc }
+      data[:vrf] = get_group_id(group_name) unless group_name.nil? || group_name.empty?
+      response = @api_resource.post('ipam/ip-addresses/', data.to_json)
+      return nil if response.code == '201'
+      { error: "Unable to add #{address} in External IPAM server" }
+    end
+
+    def delete_ip_from_subnet(ip, params)
+      group_name = params[:group_name]
+      params = { address: ip }
+      params[:vrf_id] = get_group_id(group_name) unless group_name.nil? || group_name.empty?
+      response = @api_resource.get("ipam/ip-addresses/", params)
+      json_body = JSON.parse(response.body)
+      return { error: ERRORS[:no_ip] } if json_body['count'].zero?
+      address_id = json_body['results'][0]['id']
+      response = @api_resource.delete("ipam/ip-addresses/#{address_id}/")
+      return nil if response.code == '204'
+      { error: "Unable to delete #{ip} in External IPAM server" }
+    end
+
+    def get_next_ip(mac, cidr, group_name)
+      subnet = get_ipam_subnet(cidr, group_name)
+      raise ERRORS[:no_subnet] if subnet.nil?
+      params = { limit: 1 }
+      response = @api_resource.get("ipam/prefixes/#{subnet[:id]}/available-ips/", params)
+      json_body = JSON.parse(response.body)
+      return nil if json_body.empty?
+      ip = json_body[0]['address'].split('/').first
+      cache_next_ip(@ip_cache, ip, mac, cidr, subnet[:id], group_name)
+    end
+
+    def groups_supported?
+      true
+    end
+
+    def authenticated?
+      !@token.nil?
+    end
+
+    private
+
+    def subnet_from_result(result)
+      {
+        subnet: result['prefix'].split('/').first,
+        mask: result['prefix'].split('/').last,
+        description: result['description'],
+        id: result['id'],
+      }
+    end
+
+    def subnets_from_result(result)
+      result.map do |subnet|
+        {
+          subnet: subnet['prefix'].split('/').first,
+          mask: subnet['prefix'].split('/').last,
+          description: subnet['description'],
+          id: subnet['id'],
+        }
+      end
```

Comment:
Isn't this the same?
```suggestion
      result.map { |subnet| subnet_from_result(subnet) }
```